[BUG] Panics on new DeepCopy methods

**Software version numbers**
State the version numbers of applications involved in the bug.

* Kubernetes version: N/A
* Kubernetes platform (if applicable; ex., EKS, GKE, OpenShift): N/A
* Kyverno version: latest commit https://github.com/kyverno/kyverno/commit/57d32e4883230294f0a8f27bd324871c97cd32a0

**Describe the bug**
Kyverno panics on the latest build.

**To Reproduce**
Steps to reproduce the behavior:
1. Build against the latest commit with this change https://github.com/kyverno/kyverno/pull/2651
2. install Kyverno using the above image
3. install policies from `test/best-practices`
4. see panic beblow

```
I1102 19:36:20.623994       1 policy_controller.go:200] PolicyController "msg"="policy created"  "kind"="ClusterPolicy" "name"="disallow-sysctls" "uid"="9bccd28a-38b9-4279-8100-2a2bdcbbd4f3"
E1102 19:36:20.624367       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 5042 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1af4260, 0x35f4360})
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/runtime/runtime.go:74 +0x85
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0019a9950})
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/runtime/runtime.go:48 +0x75
panic({0x1af4260, 0x35f4360})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/kyverno/kyverno/api/kyverno/v1.jsonDeepCopy({0x1ad6fc0, 0xc0041bfbf0})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/utils.go:173 +0x2f3
github.com/kyverno/kyverno/api/kyverno/v1.jsonDeepCopy({0x1ad6fc0, 0xc0041bfbc0})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/utils.go:173 +0x2f3
github.com/kyverno/kyverno/api/kyverno/v1.jsonDeepCopy({0x1ad6fc0, 0xc0041bfb90})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/utils.go:173 +0x2f3
github.com/kyverno/kyverno/api/kyverno/v1.(*Validation).DeepCopyInto(0xc0009f8610, 0xc0019a99a0)
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/utils.go:244 +0xfb
github.com/kyverno/kyverno/api/kyverno/v1.(*Validation).DeepCopy(...)
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/zz_generated.deepcopy.go:805
github.com/kyverno/kyverno/pkg/policymutation.createRuleMap({0xc0005b4000, 0x1, 0x1d95b12})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policymutation/policymutation.go:485 +0x385
github.com/kyverno/kyverno/pkg/policymutation.generateRulePatches({{{0xc0002a7e10, 0xd}, {0xc0002a7e00, 0xd}}, {{0xc0002a7ed0, 0x10}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...}, ...)
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policymutation/policymutation.go:510 +0x8c
github.com/kyverno/kyverno/pkg/policymutation.GeneratePodControllerRule({{{0xc0002a7e10, 0xd}, {0xc0002a7e00, 0xd}}, {{0xc0002a7ed0, 0x10}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...}, ...)
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policymutation/policymutation.go:382 +0x418
github.com/kyverno/kyverno/pkg/policymutation.GenerateJSONPatchesForDefaults(0xc0044a7340, {0x262eb38, 0xc00084e3c0})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policymutation/policymutation.go:46 +0x634
github.com/kyverno/kyverno/pkg/kyverno/common.MutatePolicy(0xc0044a7340, {0x262eb38, 0xc00084e3c0})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/kyverno/common/common.go:271 +0x45
github.com/kyverno/kyverno/pkg/policy.(*PolicyController).addPolicy(0xc000698780, {0x1d631c0, 0xc0044a7340})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policy/policy_controller.go:208 +0x34e
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
        /Users/shutingzhao/go/pkg/mod/k8s.io/client-go@v0.21.4/tools/cache/controller.go:231
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /Users/shutingzhao/go/pkg/mod/k8s.io/client-go@v0.21.4/tools/cache/shared_informer.go:777 +0x9f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f9317e8b208)
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000c40f38, {0x25d3800, 0xc00197f620}, 0x1, 0xc000a7f920)
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000054c90, 0x3b9aca00, 0x0, 0x80, 0x117b240)
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000d95180)
        /Users/shutingzhao/go/pkg/mod/k8s.io/client-go@v0.21.4/tools/cache/shared_informer.go:771 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:73 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:71 +0x88
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x13bd8f3]

goroutine 5042 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0019a9950})
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x1af4260, 0x35f4360})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/kyverno/kyverno/api/kyverno/v1.jsonDeepCopy({0x1ad6fc0, 0xc0041bfbf0})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/utils.go:173 +0x2f3
github.com/kyverno/kyverno/api/kyverno/v1.jsonDeepCopy({0x1ad6fc0, 0xc0041bfbc0})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/utils.go:173 +0x2f3
github.com/kyverno/kyverno/api/kyverno/v1.jsonDeepCopy({0x1ad6fc0, 0xc0041bfb90})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/utils.go:173 +0x2f3
github.com/kyverno/kyverno/api/kyverno/v1.(*Validation).DeepCopyInto(0xc0009f8610, 0xc0019a99a0)
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/utils.go:244 +0xfb
github.com/kyverno/kyverno/api/kyverno/v1.(*Validation).DeepCopy(...)
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/api/kyverno/v1/zz_generated.deepcopy.go:805
github.com/kyverno/kyverno/pkg/policymutation.createRuleMap({0xc0005b4000, 0x1, 0x1d95b12})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policymutation/policymutation.go:485 +0x385
github.com/kyverno/kyverno/pkg/policymutation.generateRulePatches({{{0xc0002a7e10, 0xd}, {0xc0002a7e00, 0xd}}, {{0xc0002a7ed0, 0x10}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...}, ...)
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policymutation/policymutation.go:510 +0x8c
github.com/kyverno/kyverno/pkg/policymutation.GeneratePodControllerRule({{{0xc0002a7e10, 0xd}, {0xc0002a7e00, 0xd}}, {{0xc0002a7ed0, 0x10}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...}, ...)
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policymutation/policymutation.go:382 +0x418
github.com/kyverno/kyverno/pkg/policymutation.GenerateJSONPatchesForDefaults(0xc0044a7340, {0x262eb38, 0xc00084e3c0})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policymutation/policymutation.go:46 +0x634
github.com/kyverno/kyverno/pkg/kyverno/common.MutatePolicy(0xc0044a7340, {0x262eb38, 0xc00084e3c0})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/kyverno/common/common.go:271 +0x45
github.com/kyverno/kyverno/pkg/policy.(*PolicyController).addPolicy(0xc000698780, {0x1d631c0, 0xc0044a7340})
        /Users/shutingzhao/go/src/github.com/realshuting/kyverno/pkg/policy/policy_controller.go:208 +0x34e
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
        /Users/shutingzhao/go/pkg/mod/k8s.io/client-go@v0.21.4/tools/cache/controller.go:231
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /Users/shutingzhao/go/pkg/mod/k8s.io/client-go@v0.21.4/tools/cache/shared_informer.go:777 +0x9f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f9317e8b208)
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000c40f38, {0x25d3800, 0xc00197f620}, 0x1, 0xc000a7f920)
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000054c90, 0x3b9aca00, 0x0, 0x80, 0x117b240)
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000d95180)
        /Users/shutingzhao/go/pkg/mod/k8s.io/client-go@v0.21.4/tools/cache/shared_informer.go:771 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:73 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /Users/shutingzhao/go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:71 +0x88
q%                                                                                                                                                            

```

--- a/api/kyverno/v1/utils.go
+++ b/api/kyverno/v1/utils.go
@@ -162,15 +162,23 @@ func jsonDeepCopy(in apiextensions.JSON) *apiextensions.JSON {
 		out_tmp := make([]interface{}, len(in))
 
 		for i, v := range in {
-			out_tmp[i] = *jsonDeepCopy(v)
+			if v != nil {
+				out_tmp[i] = *jsonDeepCopy(v)
+			} else {
+				out_tmp[i] = nil
+			}
 		}
 
 		*out = out_tmp
 	case map[string]interface{}:
 		out_tmp := make(map[string]interface{})
 
-		for i, v := range in {
-			out_tmp[i] = *jsonDeepCopy(v)
+		for k, v := range in {
+			if v != nil {
+				out_tmp[k] = *jsonDeepCopy(v)
+			} else {
+				out_tmp[k] = nil
+			}
 		}
 
 		*out = out_tmp

